### PR TITLE
replace sklearn assert function with pytest and changes.rst

### PR DESCRIPTION
**Pull request by amy12xx**
Changes:
- Replaced sklearn's assert_raise_message() with pytest raises in target and similarity encoder.
- update to changes.rst for minhash handle_missing functionality.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -1,3 +1,9 @@
+Release 0.0.8
+=============
+
+* **MinHashEncoder**: Added a "handle_missing" attribute to allow encoding
+  with missing values.
+
```
Comment (GaelVaroquaux) [inline CHANGES.rst:6]: I think that I released 0.0.7 after you added this, hence it should be added to the 0.0.7.

**Review by GaelVaroquaux — COMMENTED** [review]
Thanks a lot. A tiny comment, but aside from that it's a great addition and it's ready to merge.

File: CHANGES.rst (hunk 1 above)
Comment (amy12xx) [reply CHANGES.rst]: Yes, I wasn't too sure about that. I've fixed it.

**Review by GaelVaroquaux — APPROVED** [review]
LGTM. Merging. Thanks a lot!!

**Pull request merged**